Source/Evolve/WarpXEvolve.cpp

```diff
@@ -341,6 +326,21 @@ WarpX::Evolve (int numsteps)
             early_params_checked = true;
         }
 
+        // warpx_py_afterstep runs with the updated global time. It is included
+        // in the evolve timing.
+        if (warpx_py_afterstep) warpx_py_afterstep();
```

Comment:
Maybe we want to keep this before the diagnostics, to keep the current code behavior?